### PR TITLE
Cleanup code

### PR DESCRIPTION
**Pull request by MichaelDeBoey**
- Add default value of empty object (`{}`) instead of `null` for `data` (just like with [`Inertia.visit`](https://github.com/inertiajs/inertia/blob/master/src/inertia.js#L45))
- Add default value of `noop` for `onClick` so there's no need for an existence check
- Inline `onClick` just like in [`inertia-vue`](https://github.com/inertiajs/inertia-vue/blob/8e418a8cf23514250c62d156249e560a20125321/src/link.js#L32-L46)
- Little nitpicks

### CODE REVIEW

File: src/Link.js [hunk 1]
```diff
@@ -1,32 +1,33 @@
...
-  children,
-  onClick,
   ...props
 }) {
-  const visit = useCallback(event => {
-    if (onClick) {
+  return createElement('a', {
+    ...props,
+    href,
+    onClick: useCallback(event => {
```
Comment (sebastiandedeyne) [inline src/Link.js]: I'd rather keep all hooks on top of a component so they're explicit. If someone would add an early return before `createElement` we'd be introducing a silent bug, so I'd rather revert this change.

File: src/Link.js [hunk 2]
```diff
@@ -1,32 +1,33 @@
 import Inertia, { shouldIntercept } from 'inertia'
 import { createElement, useCallback } from 'react'
 
+const noop = () => undefined
```
Comment (sebastiandedeyne) [inline src/Link.js]: This one's good. Please add an empty line between `noop` and the export.

File: src/Link.js (hunk 1 above)
Comment (MichaelDeBoey) [reply src/Link.js]: I did this because @reinink likes to have everything inline (see https://github.com/inertiajs/inertia-vue/pull/21#discussion_r282066436)

Comment (sebastiandedeyne) [reply src/Link.js]: Trying to keep things consistent is a good target, but avoiding footguns in the current environment is more valuable. Vue doesn't have the same caveats as hooks rules.

Comment (MichaelDeBoey) [reply src/Link.js]: Reverted the commit 🙂

**Comment by MichaelDeBoey** [comment]
Resolved all change requests 🙂

**Comment by sebastiandedeyne** [comment]
Thanks!

**Comment by MichaelDeBoey** [comment]
With pleasure  🙂
